wandb/wandb_run.py

```diff
@@ -440,38 +452,43 @@ def __repr__(self):
 
     @property
     def name(self):
-        """We assume the first line of the description is the name users
-        want to use, and we automatically set it to id if the user didn't specify
-        """
-        if self.run_name is not None:
-            return self.run_name
-        return self.name_and_description.split("\n")[0]
+        if self._name is not None:
+            return self._name
+        return self._name_and_description.split("\n")[0]
 
     @name.setter
     def name(self, name):
-        self.run_name = name
-        parts = self.name_and_description.split("\n", 1)
+        self._name = name
+        parts = self._name_and_description.split("\n", 1)
         parts[0] = name
-        self.name_and_description = "\n".join(parts)
+        self._name_and_description = "\n".join(parts)
+
+    @property
+    def display_name(self):
+        """For compatibility with the public API."""
+        return self.name
 
-    # deprecate description in future release in favor of notes
     @property
     def description(self):
-        parts = self.name_and_description.split("\n", 1)
+        warnings.warn('Run.description is deprecated. Please use run.notes instead.', DeprecationWarning)
```

Comment:
I added a `termwarn` method for warnings...  I suppose ideally that would use warnings.warn behind the scenes so folks could turn it off.

Reply:
Hmm.. yeah I dunno how that would happen. I can switch to termwarn.

Reply:
Fixed.